src/commitlint/index.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { Tree, VirtualTree } from '@angular-devkit/schematics';
+import {
+  SchematicTestRunner,
+  UnitTestTree
+} from '@angular-devkit/schematics/testing';
+import * as path from 'path';
+
+const collectionPath = path.join(__dirname, '../collection.json');
+
+describe('@co-it/schematics:commitlint', () => {
+  let runner: SchematicTestRunner;
+  let treeBefore: Tree;
+
+  beforeEach(() => {
+    runner = new SchematicTestRunner('commitlint', collectionPath);
+    treeBefore = new UnitTestTree(new VirtualTree());
+  });
+
+  describe('When commitlint is not installed', () => {
+    beforeEach(() => {
+      const packageBeforeInstall = { devDependencies: {} };
+      treeBefore.create('package.json', JSON.stringify(packageBeforeInstall));
+    });
+
+    it('should add required devDependencies', () => {
+      const treeAfter = runner.runSchematic('commitlint', {}, treeBefore);
+
+      const packageAfterInstall = JSON.parse(
+        treeAfter.readContent('package.json')
+      );
+      expect(packageAfterInstall.devDependencies).toEqual(
+        expect.objectContaining({
+          '@commitlint/cli': 'latest',
```

Comment:
We need to take care of the `latest` version.
I will create another issue for it.